File: tool/table/client/fighteffect.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let fighteffect_map = null;
export function fighteffect_map_init(config_obj:Object):void{
	fighteffect_map = config_obj["fighteffect_map"];
}


export class Fighteffect extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = fighteffect_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Fighteffect(key){
		if(Fighteffect.m_static_map.hasOwnProperty(key) == false){
			Fighteffect.m_static_map[key] = Fighteffect.create_Fighteffect(key);
		}
		return Fighteffect.m_static_map[key];	
	}
	public static create_Fighteffect(key){
		if(fighteffect_map.hasOwnProperty(key)){
			return new Fighteffect(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return fighteffect_map;
	}
}
}

File: app/config/fighteffect.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


fighteffect_map = {};
fighteffect_map[1001] = {"id":1001,"name":"复活",};
fighteffect_map[1006] = {"id":1006,"name":"吸血",};
fighteffect_map[1038] = {"id":1038,"name":"清除buff",};


class Fighteffect:
	def __init__(self, key):
		config = fighteffect_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Fighteffect(key):
		config = fighteffect_map.get(key);
		if not config:
			return
		return Fighteffect(key)

